lib/_http_agent.js

```diff
@@ -168,8 +168,10 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     // we have a free socket, so use that.
     var socket = this.freeSockets[name].shift();
     // Assign the handle a new asyncId and run any init() hooks.
-    socket._handle.asyncReset();
-    socket[async_id_symbol] = socket._handle.getAsyncId();
+    if (socket._handle) {
```

Comment:
I don't think this check is necessary. It's only the socket creation that's an issue.